core/src/models/Property.ts

```diff
@@ -393,6 +401,48 @@ export class Property extends LoggedModel<Property> {
     };
   }
 
+  // --- Cache Methods --- //
+
+  static async findAllWithCache(): Promise<Property[]> {
+    const now = new Date().getTime();
+    if (
+      _cachedProperties.expires > now &&
+      _cachedProperties.properties.length > 0
+    ) {
+      return _cachedProperties.properties;
+    }
+
+    _cachedProperties.properties = await Property.findAll();
+    _cachedProperties.expires = now + CACHE_TTL;
+    return _cachedProperties.properties;
+  }
+
+  static async findOneWithCache(key: string, value: string) {
```

Comment:
If this is always "id", it might be nice to make a method for that `findByPkWithCache()` or something

Reply:
or we can switch the oder and make a default -98c3ac24fd41a019890622023bc0994982f4c530

Reply:
Works for me.